### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by Prodge**
Test framework: Mocha
Assertion framework: Chai
CI: Travis

Tests are in /test folder.

Route tests are broken into seperate files with a common setup/teardown.

We currently have 60 tests which give 70.8% coverage.

The server has been moved into its own file in /app/server.js. the start.js file requires this file and starts the server. 
This allows the tests to require that file and start the server on a seperate port.
The same applies to mongoose. 

Start the server with: npm start
Run tests with: npm test
Run coverage with: npm run coverage

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+language:
```
Comment (Prodge) [inline .travis.yml]: This is what Travis uses to create our test environment

File: root/app/models/user.js [hunk 2]
```diff
@@ -3,44 +3,44 @@ var Schema = mongoose.Schema;
 var bcrypt = require('bcrypt');
 
 var UserSchema = new Schema({
-    name: {
-        type: String,
-        required: true,
-        unique: true,
-    },
+  name: {
+    type: String,
```
Comment (Prodge) [inline root/app/models/user.js]: retabbed

File: root/get_coverage_report.sh [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Run with name as argument
```
Comment (Prodge) [inline root/get_coverage_report.sh]: Use this script to pull the coverage report from the server for local viewing

run as: sh get_coverage_report.sh wimo

File: root/package.json [hunk 4]
```diff
@@ -3,24 +3,30 @@
   "version": "0.0.1",
   "private": true,
   "scripts": {
-    "start": "node app.js"
+    "start": "node start.js",
+    "test": "mocha --recursive",
+    "coverage": "istanbul cover _mocha -- --recursive"
```
Comment (Prodge) [inline root/package.json]: We can just use these npm commands now :)

File: root/show_logs.sh [hunk 5]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+# Checks if there are any debug outputs present in the codebase
```
Comment (Prodge) [inline root/show_logs.sh]: Use this to check if we have left any console.log() statements around

File: root/start.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+var server = require('./app/server');
```
Comment (Prodge) [inline root/start.js]: Starts the server and database

**Comment by wimoMisterX** [comment]
Well done!!!!
Very AWesome :DD:D

**Pull request merged**